src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -4231,7 +4231,7 @@ public bool ConnectAsync(SocketAsyncEventArgs e)
                     {
                         InternalBind(new IPEndPoint(IPAddress.Any, 0));
                     }
-                    else
+                    else if (endPointSnapshot.AddressFamily == AddressFamily.InterNetworkV6)
```

Comment:
If I add a check for AF.Unix, then the behavior should be in-line with Desktop. On Windows the unix domain socket family should fail in Socket ctor on CoreFX and on Desktop too, it won't make it to this line. I'll run a small app on desktop to check that.